src/java.desktop/macosx/native/libosxapp/ThreadUtilities.m

```diff
@@ -31,6 +31,7 @@
 
 // The following must be named "jvm", as there are extern references to it in AWT
 JavaVM *jvm = NULL;
+static BOOL isNSApplicationOwner = YES;
```

Comment:
One possible suggestion would be to flip the default to `NO`, so as not to cache the env until we know we are the owner.

Reply:
Flipped the default to NO..